hudi-common/src/test/java/org/apache/hudi/common/model/TestOverwriteNonDefaultsWithLatestAvroPayload.java

```diff
@@ -55,20 +57,23 @@ public void testActiveRecords() throws IOException {
     record1.put("ts", 0L);
     record1.put("_hoodie_is_deleted", false);
     record1.put("city", "NY0");
+    record1.put("child", Arrays.asList("A"));
 
     GenericRecord record2 = new GenericData.Record(schema);
     record2.put("id", "2");
     record2.put("partition", "");
     record2.put("ts", 1L);
     record2.put("_hoodie_is_deleted", false);
     record2.put("city", "NY");
+    record2.put("child", Collections.emptyList());
 
     GenericRecord record3 = new GenericData.Record(schema);
     record3.put("id", "2");
     record3.put("partition", "partition1");
     record3.put("ts", 1L);
     record3.put("_hoodie_is_deleted", false);
     record3.put("city", "NY0");
+    record3.put("child", Arrays.asList("A"));
```

Comment:
Can you confirm that with array data type, this test was failing if not the fix currently given in the patch? LGTM. can land it once you confirm this.

Reply:
After pushing the latest commit, test ran twice and the failure module of each is different. Therefore, the reason of test failure is independence of this patch

Reply:
Can we land this?